frontend/packages/akr/src/redux/selectors/clerkTranslator.ts

```diff
@@ -90,20 +90,36 @@ export const selectFilteredSelectedTranslators = createSelector(
 );
 
 // Helpers
+
 const filterByAuthorisationStatus = (
   translator: ClerkTranslator,
   status: AuthorisationStatus,
   currentDate: Dayjs,
   expiringSoonDate: Dayjs
 ) => {
-  return translator.authorisations.find((a) =>
-    matchesAuthorisationStatus(
-      { authorisationStatus: status },
-      currentDate,
-      expiringSoonDate,
-      a
-    )
-  );
+  if (status === AuthorisationStatus.Expired) {
+    // If selected AuthorisationStatus is Expired, we must search all expired authorisations to check
+    // that there are no corresponding effective authorisations for the same language pair.
+    return (
+      filterAuthorisationsByStatus(
+        translator.authorisations,
+        status,
+        currentDate,
+        expiringSoonDate
+      ).length > 0
+    );
+  } else {
+    // Otherwise, we may short circuit the evaluation upon finding the first authorisation
+    // matching the desired authorisation status.
+    return translator.authorisations.find((a) =>
+      matchesAuthorisationStatus(
+        { authorisationStatus: status },
+        currentDate,
+        expiringSoonDate,
+        a
+      )
+    );
+  }
 };
 
 const filterByAuthorisationCriteria = (
```

Comment:
Pitäisikö laittaa funktiolle `filterByAuthorisationStatus`, `filterByAuthorisationCriteria`, `filterAuthorisationsByStatus` vielä funktio tason kommentit siitä, että mitä ne tekee? Lisäksi kannattaisiko siirtää `filterAuthorisationsByStatus` authorisation selectoriin? Ei se välttämättä lisää mitään arvoa :)

Reply:
```
/* selityksiä mitä funktiot matches tekee. Rittää varmaan yksi ylä tason kommentti?*/
matches*
...

/*  selityksiä mitä funktiot filterBy tekee. Rittää varmaan yksi ylä tason kommentti? */
filterBy*
....

/* ...*/
filterAuthorizationBy*
...
```

Auttaisiko esim yllä oleva järjestys.

Reply:
Tää voisi olla ihan toimiva vaihtoehto. Toinen mahdollisuus olisi ehkä nimetä näitä funkkareita toisin:
`filterByX` -> `translatorMatchesX`, `matchesX` -> `authorisationMatchesX`. Olisiko tämä vielä selkeämpi?

Reply:
joo, kuulostaa hyvältä.

Reply:
Muutettu nyt.